.github/workflows/weekly.yml

```diff
@@ -7,38 +7,17 @@ on:
 
 jobs:
   weekly:
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ${{ matrix.os }}
     strategy:
-      max-parallel: 5
       fail-fast: false
       matrix:
-        os: [ubuntu]
-        python: [3.7, 3.8]
-        extras: ['']
-        include:
-          - os: macos
-            python: 3.8
-          - os: windows
-            python: 3.8
-          - os: ubuntu
-            python: 3.9
-            extras: -minimal
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
+        os: [windows-latest, ubuntu-latest, macos-latest]
 
     steps:
     - uses: actions/checkout@v2
-    - uses: conda-incubator/setup-miniconda@v2
+    - uses: astral-sh/setup-uv@v5
```

Comment:
Have you confirmed that uv can handle installing the ecCodes library in addition to the ecCodes python package?